doc/examples/applications/plot_3d_structure_tensor.py

```diff
@@ -0,0 +1,160 @@
+from skimage import (
+    data, feature
+)
+
+
+#####################################################################
+# Load image
+# ==========
+# This biomedical image is available through `scikit-image`'s data registry.
+
+data = data.kidney()
+
+#####################################################################
+# What exactly are the shape and size of our 3D multichannel image?
+
+print(f'number of dimensions: {data.ndim}')
+print(f'shape: {data.shape}')
+print(f'dtype: {data.dtype}')
+
+#####################################################################
+# For the purposes of this tutorial, we shall consider only the second colour
+# channel, which leaves us with a 3D single-channel image. What is the range
+# of values?
+
+n_plane, n_Y, n_X, n_chan = data.shape
+v_min, v_max = data[:, :, :, 1].min(), data[:, :, :, 1].max()
+print(f'range: ({v_min}, {v_max})')
+
+#####################################################################
+# Let us visualize the middle slice of our 3D image.
+
+px.imshow(
+    data[n_plane // 2, :, :, 1],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'}
+)
+
+#####################################################################
+# Let us pick a specific region, which shows relative X-Y isotropy. In
+# contrast, the gradient is quite different (and, for that matter, weak) along
+# Z.
+
+sample = data[5:13, 380:410, 370:400, 1]
+step = 3
+cols = sample.shape[0] // step + 1
+_, axes = plt.subplots(nrows=1, ncols=cols, figsize=(16, 8))
+
+for it, (ax, image) in enumerate(zip(axes.flatten(), sample[::step])):
+    ax.imshow(image, cmap='gray', vmin=v_min, vmax=v_max)
+    ax.set_title(f'Plane = {5 + it * step}')
+    ax.set_xticks([])
+    ax.set_yticks([])
+
+#####################################################################
+# To view the sample data in 3D, run the following code:
+#
+# .. code-block:: python
+#
+#     import plotly.graph_objects as go
+#
+#     (n_Z, n_Y, n_X) = sample.shape
+#     Z, Y, X = np.mgrid[:n_Z, :n_Y, :n_X]
+#
+#     fig = go.Figure(
+#         data=go.Volume(
+#             x=X.flatten(),
+#             y=Y.flatten(),
+#             z=Z.flatten(),
+#             value=sample.flatten(),
+#             opacity=0.5,
+#             slices_z=dict(show=True, locations=[4])
+#         )
+#     )
+#     fig.show()
+
+#####################################################################
+# Compute structure tensor
+# ========================
+# About the brightest region (i.e., at X ~ 22 and Y ~ 17), we can see
+# variations (and, hence, strong gradients) along either X or Y over a typical
+# length of 3.
+
+px.imshow(
+    sample[0, :, :],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'},
+    title='Interactive view of bottom slice of sample data.'
+)
+
+#####################################################################
+# Therefore, we choose a 'width' of 3 for the window function.
+
+sigma = 3
```

Comment:
sigma of 3 seems pretty large, particularly for the first axis which is only size 8 here. To make the result less dependent on boundary conditions, I would use something a bit smaller (maybe 1.0-1.5 or so?)

This is the Gaussian kernel that would be used by SciPy's `gaussian_filter` for sigma=3:
```Python
from scipy.ndimage.filters import _gaussian_kernel1d
sigma = 3
truncate = 4
kernel = _gaussian_kernel1d(sigma, order=0, radius=int(truncate*sigma + 0.5))
plt.figure(); plt.plot(kernel, 'k.-')
```
![g](https://user-images.githubusercontent.com/6528957/105540639-86368f00-5cc4-11eb-9080-6958643520f6.png)


also, The `structure_tensor` docstring indicates `sigma` should be a float, but it looks like the function can accept a unique sigma per axis (e.g. `sigma=(3, 3, 1)`). If that is intentional, we should update the docstring to mention it.

Reply:
Right, fixed in 521d4d235e86ec64e709e586fc864850b45044fb.

> The structure_tensor docstring indicates sigma should be a float, but it looks like the function can accept a unique sigma per axis (e.g. sigma=(3, 3, 1)). If that is intentional, we should update the docstring to mention it.

PS: Nice find! TODO...

Reply:
Done over at #5196.